recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -39,88 +39,152 @@ Cu.importGlobalProperties(["fetch"]);
 this.EXPORTED_SYMBOLS = ["RecipeRunner"];
 
 const log = LogManager.getLogger("recipe-runner");
-const prefs = Services.prefs.getBranch("extensions.shield-recipe-client.");
 const TIMER_NAME = "recipe-client-addon-run";
-const RUN_INTERVAL_PREF = "run_interval_seconds";
-const FIRST_RUN_PREF = "first_run";
 const PREF_CHANGED_TOPIC = "nsPref:changed";
 
+const TELEMETRY_ENABLED_PREF = "toolkit.telemetry.unified";
+
+const SHIELD_PREF_PREFIX = "extensions.shield-recipe-client";
+const RUN_INTERVAL_PREF = `${SHIELD_PREF_PREFIX}.run_interval_seconds`;
+const FIRST_RUN_PREF = `${SHIELD_PREF_PREFIX}.first_run`;
+const SHIELD_ENABLED_PREF = `${SHIELD_PREF_PREFIX}.enabled`;
+const DEV_MODE_PREF = `${SHIELD_PREF_PREFIX}.dev_mode`;
+const API_URL_PREF = `${SHIELD_PREF_PREFIX}.api_url`;
+const LAZY_CLASSIFY_PREF = `${SHIELD_PREF_PREFIX}.experiments.lazy_classify`;
+
+const PREFS_TO_WATCH = [
+  RUN_INTERVAL_PREF,
+  TELEMETRY_ENABLED_PREF,
+  SHIELD_ENABLED_PREF,
+  API_URL_PREF,
+];
+
 this.RecipeRunner = {
   async init() {
-    if (!this.checkPrefs()) {
-      return;
-    }
+    this.enabled = null;
+    this.checkPrefs();  // sets this.enabled
+    this.watchPrefs();
+
+    // Run if enabled immediately on first run, or if dev mode is enabled.
+    const firstRun = Services.prefs.getBoolPref(FIRST_RUN_PREF);
+    const devMode = Services.prefs.getBoolPref(DEV_MODE_PREF);
 
-    // Run immediately on first run, or if dev mode is enabled.
-    if (prefs.getBoolPref(FIRST_RUN_PREF) || prefs.getBoolPref("dev_mode")) {
+    if (this.enabled && (devMode || firstRun)) {
       await this.run();
-      prefs.setBoolPref(FIRST_RUN_PREF, false);
     }
+    if (firstRun) {
+      Services.prefs.setBoolPref(FIRST_RUN_PREF, false);
+    }
+  },
 
+  enable() {
+    if (this.enabled) {
+      return;
+    }
+    this.enabled = true;
```

Comment:
Minor: We should update this at the end of the function in case the changes throw and we're not actually enabled.

Reply:
We could probably use a test that multiple calls to enable/disable do not register multiple watchers.